symfit/core/models.py

```diff
@@ -496,10 +499,20 @@ def __init__(self, model, independent_vars=None, params=None, **kwargs):
             self.connectivity_mapping = {var: set(independent_vars + params)
                                          for var in model}
         elif connectivity_mapping:
-            self.connectivity_mapping = connectivity_mapping
             if not isinstance(model, Mapping):
                 raise TypeError('Please provide the model as a mapping, '
                                 'corresponding to `connectivity_mapping`.')
+            elif connectivity_mapping.keys() != model.keys():
```

Comment:
I feel like it's cleaner to just *always* create an initial connectivity mapping for the symbolic components, and then update it with what's been provided. So no if statements involved.

Reply:
Fair point, in case the user provides (wrongful) information for the analytical components they will still get the error that way.